[Add panel flyout] Moving create new to the top of SavedObjectFinder

majagrubic opened the pull request:
## Summary

To draw more attention to the "Create New" button in add panel flyout on the dashboard, we want to move this button from the bottom to the top, as explored in this prototype:
https://www.figma.com/file/EwodWYRUIjbNvYoi9G4Hnh/Lensboard?node-id=611%3A0

This PR removes the footer in the `AddPanelFlyout` and adds `Create New` button as a child of `SavedObjectFinder` component.

<img width="829" alt="Screenshot 2020-01-31 at 10 12 47" src="https://user-images.githubusercontent.com/1937956/73531583-a82f6580-4412-11ea-96a3-ed44901ead74.png">
 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [x] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

Commit:
[Add panel flyout] Moving create new to the top of SavedObjectFinder

Commit:
[Add panel flyout] Moving create new to the top of SavedObjectFinder

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/)
* Commit: 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"}]}
PIPELINE-->

Commit:
Fixing failing unit test

majagrubic commented on src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.test.tsx:
```diff
@@ -80,45 +76,3 @@ test('createNewEmbeddable() add embeddable to container', async () => {
   });
 });
-
-test('selecting embeddable in "Create new ..." list calls createNewEmbeddable()', async () => {
```
This test makes no sense anymore and we have functional tests that are better suited to test interaction between components.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/)
* Commit: 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a

### History
* :broken_heart: [Build #23620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/) failed 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"}]}
PIPELINE-->

majagrubic commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/)
* Commit: 7246f1596ff339d87219723be4e068c51e6cd967

### History
* :broken_heart: [Build #23692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/) failed 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a
* :broken_heart: [Build #23620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/) failed 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/","number":"23795","commit":"7246f1596ff339d87219723be4e068c51e6cd967"}]}
PIPELINE-->

Commit:
Readd missing test

majagrubic commented on src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.test.tsx:
```diff
@@ -101,14 +109,14 @@ test('selecting embeddable in "Create new ..." list calls createNewEmbeddable()'
   const container = new HelloWorldContainer(input, { getEmbeddableFactory } as any);
   const onClose = jest.fn();
-  const component = mount<AddPanelFlyout>(
+  const component = mount(
```
I'm not super convinced this test is valid anymore. It tests interaction between components, which is what functional tests are better suited for. But leaving it here for now.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/)
* Commit: 59d36b6ff49d8d8fe6a393287299042d5b12267e

### History
* :green_heart: [Build #23795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/) succeeded 7246f1596ff339d87219723be4e068c51e6cd967
* :broken_heart: [Build #23692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/) failed 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a
* :broken_heart: [Build #23620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/) failed 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/","number":"23795","commit":"7246f1596ff339d87219723be4e068c51e6cd967"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/","number":"23845","commit":"59d36b6ff49d8d8fe6a393287299042d5b12267e"}]}
PIPELINE-->

lizozom commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/)
* Commit: 68e442a690e02acd60eba0f407c7552b32287f78

### History
* :green_heart: [Build #23845](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/) succeeded 59d36b6ff49d8d8fe6a393287299042d5b12267e
* :green_heart: [Build #23795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/) succeeded 7246f1596ff339d87219723be4e068c51e6cd967
* :broken_heart: [Build #23692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/) failed 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a
* :broken_heart: [Build #23620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/) failed 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/","number":"23795","commit":"7246f1596ff339d87219723be4e068c51e6cd967"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/","number":"23845","commit":"59d36b6ff49d8d8fe6a393287299042d5b12267e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/","number":"23910","commit":"68e442a690e02acd60eba0f407c7552b32287f78"}]}
PIPELINE-->

lizozom reviewed (commented):
After merging upstream, and pulling down the changes, the button doesn't do anything.
I'm also really surprised the tests are passing. :thinking:

streamich commented on src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/add_panel_flyout.test.tsx:
```diff
@@ -29,12 +29,20 @@ import { ContactCardEmbeddable } from '../../../../test_samples/embeddables/cont
 
-// eslint-disable-next-line
-import { coreMock } from '../../../../../../../../core/public/mocks';
+// @ts-ignore
```
Can we please remove this `// @ts-ignore` and use TypeScript casting instead, if needed.

streamich commented on src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/add_panel/tests/saved_object_finder_create_new.test.tsx:
```diff
@@ -0,0 +1,88 @@
+    expect(wrapper.find(EuiPopover).length).toEqual(1);
+    const menuPanel = wrapper.find(EuiContextMenuPanel);
+    expect(menuPanel.length).toEqual(1);
+    // @ts-ignore
```
Same here.

flash1293 reviewed (approved):
LGTM, it seems to work fine for me. 

@lizozom The button should open a popover with a single entry "Visualization" - works on my machine (TM)
<img width="207" alt="Screenshot 2020-02-03 at 12 02 24" src="https://user-images.githubusercontent.com/1508364/73648139-2e4ee480-467d-11ea-84b7-526bec23d88d.png">

@majagrubic  I'm not sure whether I like the fact that it opens the popover to the top, as a user I would have expected to open to the bottom. Like this it kind of looks like a tooltip, not like context menu. Just my 2 cents, cc @cchaos

lizozom commented:
@flash1293 @majagrubic sorry, it does work.
Testing this, I actually didn't realize this was the menu!
I thought it was just a tooltip, especially with the arrow pointing down on the button.

IMO such a menu has to open down, and not up. @elastic/kibana-design

lizozom reviewed (approved):
Overall LGTM
Assuming UX + ts issues are addressed

ryankeairns commented:
Thanks for making this change, Maja!

+1 to opening the popover down. Specifically, `anchorPosition="downRight"` since this is along the right edge of the screen.

Related, can we now add Lens to this menu?

flash1293 commented:
@ryankeairns Lens is part of the "Visualization" option - if you click it you will get the grid with all visualization types including Lens:
<img width="843" alt="Screenshot 2020-02-03 at 16 57 22" src="https://user-images.githubusercontent.com/1508364/73668532-4850ed00-46a6-11ea-971b-db9a9804f4c0.png">

I realize this isn't ideal because it adds another click till the user can work on their visualization, +1 for showing Lens in this menu directly. I think that's worth a second PR though.

ryankeairns commented:
@flash1293 of course, my mistake! I agree we should both considering adding it and doing so as a follow up PR. I'll create one.

maryia-lapata committed:
[NP] Move saved object modal into new platform (#56383)

* Move saved object modal into new platform

* Fix TS

* Revert "Fix TS"

This reverts commit f2f9f5e9356d0b0de8b18e9fceebf721393eacd0.

* Revert "Move saved object modal into new platform"

This reverts commit d0f0ea6e302e0b6acd7657b770946189143d187f.

# Conflicts:
#	src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

* Move save_object_save_modal

* Move show_saved_object_save_modal.tsx

* Move save_object_finder.tsx

* Remove unused export

* Pass I18nContext to showSaveModal

* Update i18n ids

* Fix map save

* Refactoring

* Load styles

* Revert importing styles

* Update snapshot

* Update snapshot

* Structural refactoring

* Fix path

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
Applying PR comments

Removing faulty rebase imports

Fixing unresolved conflict

Removing faulty merge files

Removing faulty import

Readd accidentally added file

majagrubic commented:
@elasticmachine merge upstream

Commit:
Removing unnecessary eslint-ignore

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25521/)
* Commit: 4e83f58222e29db9539b6a8563d9d9e9d7818e42
* Flaky suites:
  * kibana-firefoxSmoke

### History
* :green_heart: [Build #23910](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/) succeeded 68e442a690e02acd60eba0f407c7552b32287f78
* :green_heart: [Build #23845](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/) succeeded 59d36b6ff49d8d8fe6a393287299042d5b12267e
* :green_heart: [Build #23795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/) succeeded 7246f1596ff339d87219723be4e068c51e6cd967
* :broken_heart: [Build #23692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/) failed 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a
* :broken_heart: [Build #23620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/) failed 3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23620/","number":"23620","commit":"3f9429e2df463c61e811f8a5a3a6f53fdb0d0a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/","number":"23795","commit":"7246f1596ff339d87219723be4e068c51e6cd967"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/","number":"23845","commit":"59d36b6ff49d8d8fe6a393287299042d5b12267e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/","number":"23910","commit":"68e442a690e02acd60eba0f407c7552b32287f78"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25521/","number":"25521","commit":"4e83f58222e29db9539b6a8563d9d9e9d7818e42"}]}
PIPELINE-->

majagrubic commented:
Thanks Ryan, Liza and Joe! I've addressed your comments and fixed the popup.
<img width="174" alt="Screenshot 2020-02-10 at 10 14 50" src="https://user-images.githubusercontent.com/1937956/74151531-45a14b00-4c04-11ea-837c-d808c7e5d94f.png">

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25523/)
* Commit: 90054244862427f23af6df16061741807f3584c5

### History
* :yellow_heart: [Build #25521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25521/) was flaky 4e83f58222e29db9539b6a8563d9d9e9d7818e42
* :green_heart: [Build #23910](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/) succeeded 68e442a690e02acd60eba0f407c7552b32287f78
* :green_heart: [Build #23845](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/) succeeded 59d36b6ff49d8d8fe6a393287299042d5b12267e
* :green_heart: [Build #23795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/) succeeded 7246f1596ff339d87219723be4e068c51e6cd967
* :broken_heart: [Build #23692](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/) failed 45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23692/","number":"23692","commit":"45ac37cc9ca30f463d2dc82dabac2ed1e8b0e62a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23795/","number":"23795","commit":"7246f1596ff339d87219723be4e068c51e6cd967"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23845/","number":"23845","commit":"59d36b6ff49d8d8fe6a393287299042d5b12267e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/23910/","number":"23910","commit":"68e442a690e02acd60eba0f407c7552b32287f78"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25521/","number":"25521","commit":"4e83f58222e29db9539b6a8563d9d9e9d7818e42"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25523/","number":"25523","commit":"90054244862427f23af6df16061741807f3584c5"}]}
PIPELINE-->

lizozom reviewed (approved):
Thanks for fixing that UI!
Tested and LGTM
